modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -4754,7 +4762,9 @@ LineIterator& LineIterator::operator ++()
 {
     int mask = err < 0 ? -1 : 0;
     err += minusDelta + (plusDelta & mask);
-    ptr += minusStep + (plusStep & mask);
+    p.x += minusShift + (plusShift & mask);
+    p.y += minusStep + (plusStep & mask);
+    ptr = (uchar*)ptr0 + (size_t)p.y*step + (size_t)p.x*elemSize;
```

Comment:
OpenCV drawing functions are not optimized at all.
This change may bring unnecessary performance degradation to existed functionality.

---

It is impossible to run UBSAN on that code (adding offset to NULL pointer is an ERROR). Adding one more check is performance degradation again.

---

See comments in previous patch about that: #13843 (**mentioned** in PR description)

Reply:
ok, I've eliminated the performance degradation and also made the code UBSAN compatible